core/src/main/java/org/apache/iceberg/encryption/EncryptionManagerFactory.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.encryption;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Map;
+import org.apache.iceberg.TableMetadata;
+
+public interface EncryptionManagerFactory extends Closeable {
```

Comment:
Do you have an example of how this is used in context? I want to make sure what we're adding to catalogs or table implementations makes sense.

Reply:
Sure, in https://github.com/apache/iceberg/pull/5544/files 

( eg https://github.com/apache/iceberg/pull/5544/files#diff-dcb8898b53c9729e27fea26093e90a05e86cc2682a03210cf5f56fd5a260f174R237 and https://github.com/apache/iceberg/pull/5544/files#diff-b840ad8566560883268ebb23aa1de123a05599a1aeeda31bf66dfc0cb54a7e82R194 )

Reply:
Basically, it's an object in TableOperations, that allows up to create an EncryptionManager from both catalog properties and table metadata.